packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -593,14 +593,15 @@ export interface OptionalFieldEditor extends FieldEditor<OptionalChangeset> {
 			fill: ChangesetLocalId;
 			detach: ChangesetLocalId;
 		},
+		revision: RevisionTag,
```

Comment:
Rather than taking this additonal `revision` argument, I think it would make more sense to update `fill` and `detach` to be `ChangeAtomId`s instead of `ChangesetLocalId`s. It problably should have taken `ChangeAtomId`s from the start (with an undefined revision). My reasoning here is that the optional field FieldKind doesn't really have the idea of establishing a 1:1 correspondence between a changeset and a revision tag: a single optional changeset can represent a composition of changes made at different revisions. The only thing the optional field FieldKind cares about is precicely identifying is registers, which it does using `ChangeAtomId`s (and the "self" string).